Source/scrollrt.cpp

```diff
@@ -696,7 +696,7 @@ static void scrollrt_draw_dungeon(int sx, int sy, int dx, int dy)
 	bDead = dDead[sx][sy];
 	bMap = dTransVal[sx][sy];
 
-	negMon = dMonster[sx][sy - 1];
+	negMon = sy > 0 ? dMonster[sx][sy - 1] : 0;

```

Comment:
This ternary expression is a bit magic, perhaps a comment would help give insight albeit at the loss of *smoke and mirrors* :)

If the screen Y position is 0 or below, then don't draw the monster? Was this bugged in the original Diablo rendering, but never visible as the panel would always overlap the monster?

Reply:
I believe the indexes are map row/columns, this just checks for possible out of bounds

Reply:
Ah, you are correct. Did we ever run into such an overflow in Vanilla?

Reply:
no idea :p the game probably wouldn't care :D

Reply:
Afaik this is only a problem at high resolution because you can end up rendering parts that are outside the game world. Blizzard added a 16 tile border around the world filled with dirt to avoid this kind of issue. Probably a better solution would have been to simply have the render draw dirt if rendering parts that are outside the dungion.